wordpress-mobile/WordPress-FluxC-Android
# [Coupons] Fetch and observe a coupon

Fixes #2336, which adds a method to fetch and observe a single coupon.

**To test:**
Run the `CouponDaoTests` tests.
+
1. Start the example app
2. Sign in & go to Woo -> Coupons -> Select site
3. Tap on Fetch single coupon button
4. Enter a random string
5. Notice an error message is printed
6. Tap on the button again and enter a random number
7. Notice a message is displayed that the coupon wasn't found
8. Tap on the button again and enter a valid coupon ID
9. Notice the coupon entity is printed
